NGitLab/Impl/GroupsClient.cs

```diff
@@ -36,61 +35,79 @@ public GitLabCollectionResponse<Group> GetAsync(GroupQuery query)
         return _api.Get().GetAllAsync<Group>(url);
     }
 
-    private static string CreateGetUrl(GroupQuery query)
+    public Task<(IReadOnlyCollection<Group> Page, int? Total)> PageAsync(PageQuery<GroupQuery> query, CancellationToken cancellationToken = default)
     {
-        var url = Group.Url;
+        var url = CreateGetUrl(query?.Query, query?.Page, query?.PerPage);
+        return _api.Get().PageAsync<Group>(url, cancellationToken);
+    }
+
+    private static string CreateGetUrl(GroupQuery query, int? page = null, int? perPage = null)
+    {
+        var url = Url;
+
+        url = Utils.AddPageParams(url, page, perPage);
 
-        if (query.SkipGroups != null && query.SkipGroups.Any())
+        if (query is not null)
```

Comment:
You can invert the condition and return early. That will remove one level of indentation